### PR TITLE
python311Packages.tinygrad: init at 0.7.0

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Description of changes

Initialises the tinygrad ml framework in nixpkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tinygrad/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  pythonImportsCheck = [ "tinygrad" ];
+
+  meta = with lib; {
+    description = "You like pytorch? You like micrograd? You love tinygrad";
```
Comment (natsukium) [inline pkgs/development/python-modules/tinygrad/default.nix:28]: Although this is an upstream issue, I don't think it explains much.
IMO, the title of [this page](https://tinygrad.org/), `A simple and powerful neural network framework`, would be more appropriate.

File: pkgs/development/python-modules/tinygrad/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    homepage = "https://github.com/tinygrad/tinygrad";
```
Comment (natsukium) [inline pkgs/development/python-modules/tinygrad/default.nix:29]:
```suggestion
    homepage = "https://github.com/tinygrad/tinygrad";
    changelog = "https://github.com/tinygrad/tinygrad/releases/tag/${src.rev}";
```
Adding the changelog is useful for future updates.

File: pkgs/development/python-modules/tinygrad/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools
+, wheel
+}:
+
+buildPythonPackage rec {
+  pname = "tinygrad";
+  version = "0.7.0";
+  pyproject = true;
```
Comment (natsukium) [inline pkgs/development/python-modules/tinygrad/default.nix:11]:
```suggestion
  pyproject = true;
  
  disabled = pythonOlder "3.8";
```
https://github.com/tinygrad/tinygrad/blob/8b354b3f73116e3acaad41cf55281de242709506/setup.py#L23

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by GaetanLepage** [comment]
Followed up in #287914

**Comment by MatthewCroughan** [comment]
Obsoleted by https://github.com/NixOS/nixpkgs/pull/287914

**Pull request closed without merging**